addons/web/static/src/core/l10n/utils.js

```diff
@@ -0,0 +1,74 @@
+import { user } from "@web/core/user";
+
+/**
+ * Convert Unicode TR35-49 list pattern types to ES Intl.ListFormat options
+ */
+const LIST_STYLES = {
+    standard: {
+        type: "conjunction",
+        style: "long",
+    },
+    "standard-short": {
+        type: "conjunction",
+        style: "short",
+    },
+    or: {
+        type: "disjunction",
+        style: "long",
+    },
+    "or-short": {
+        type: "disjunction",
+        style: "short",
+    },
+    unit: {
+        type: "unit",
+        style: "long",
+    },
+    "unit-short": {
+        type: "unit",
+        style: "short",
+    },
+    "unit-narrow": {
+        type: "unit",
+        style: "narrow",
+    },
+};
+
+/**
+ * Format the items in `list` as a list in a locale-dependent manner with the chosen style.
+ * 
+ * The available styles are defined in the Unicode TR35-49 spec:
+ * * standard:
+ *   A typical "and" list for arbitrary placeholders.
+ *   eg. "January, February, and March"
+ * * standard-short:
+ *   A short version of an "and" list, suitable for use with short or abbreviated placeholder values.
+ *   eg. "Jan., Feb., and Mar."
+ * * or:
+ *   A typical "or" list for arbitrary placeholders.
+ *   eg. "January, February, or March"
+ * * or-short:
+ *   A short version of an "or" list.
+ *   eg. "Jan., Feb., or Mar."
+ * * unit:
+ *   A list suitable for wide units.
+ *   eg. "3 feet, 7 inches"
+ * * unit-short:
+ *   A list suitable for short units
+ *   eg. "3 ft, 7 in"
+ * * unit-narrow:
+ *   A list suitable for narrow units, where space on the screen is very limited.
+ *   eg. "3′ 7″"
+ * 
+ * See https://www.unicode.org/reports/tr35/tr35-49/tr35-general.html#ListPatterns for more details.
+ * 
+ * @param {string[]} list The array of values to format into a list.
+ * @param {"standard"|"standard-short"|"or"|"or-short"|"unit"|"unit-short"|"unit-narrow"} style The style to format the list with.
+ * @param {string} locale_code The locale to use (i.e. en-US).
+ * @returns {string} The formatted list.
+ */
+export function formatList(list, style = "standard", locale_code = null) {
```

Comment:
I'd make locale_code an empty string by default to keep it monomorphic
```suggestion
export function formatList(list, style = "standard", locale_code = "") {
```